pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -499,6 +487,32 @@ func CreateGCECloud(config *CloudConfig) (*GCECloud, error) {
 	return gce, nil
 }
 
+func determineSubnetURL(service *compute.Service, networkProjectID, networkName, region string) (string, bool, error) {
```

Comment:
Under what circumstance would a return of (string, error) not be sufficient to provide the necessary data? It returns either an error, or a subnet, but never both.

```
const (
  ErrNoNetwork = fmt.Errorf("eould not retrieve network %q in attempt to determine if legacy network or see list of subnets; err: %v", networkName, err)
  ErrLegacyNetwork = fmt.Errorf("Determined network %q is of type legacy.", networkName")
  ErrSubnetList = fmt.Errorf("error listing subnets of network %q within region %v. err: %v", networkName, region, err)
  ErrNoAutoSubnets = fmt.Errorf("could not find any auto subnet in region %q within list %+v", region, subnets)
)
...

[line 442] subnetURL, err = determineSubnetURL(service, netProjID, networkName, config.Region)

  if _, ok := err.(ErrLegacyNetwork) {
    // Gracefully fail because kubelet calls CreateGCECloud without any config, and API calls will fail coming from minions.
    glog.Warningf("Could not retrieve network %q in attempt to determine if legacy network or see list of subnets, err %v", networkURL, err)
    glog.Warning(err)
  }

 ...
}
```